### PR TITLE
cmake WITH_LIBSODIUM option is broken

### PR DESCRIPTION
**Pull request by boringuy**
- Fixed the variable name in platform.hpp.in

### CODE REVIEW

File: src/ctx.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ zmq::ctx_t::~ctx_t ()
 
     //  If we've done any Curve encryption, we may have a file handle
     //  to /dev/urandom open that needs to be cleaned up.
-#ifdef ZMQ_HAVE_CURVE
+#if defined (ZMQ_USE_TWEETNACL)
```
Comment (bluca) [inline src/ctx.cpp]: are you sure about this? sodium has this function too
https://download.libsodium.org/doc/generating_random_data/

Comment (boringuy) [inline src/ctx.cpp]: Ah... I fixed this before I fix the CMAKE variable then. So, this error was cause by the same problem. I will change this back. Thanks.

**Comment by boringuy** [comment]
Ok. Fixed the pull request, it ends up to be just a one line change in platform.hpp.in

**Comment by bluca** [comment]
Thanks! For your next contributions, if you have fixup commits, please rebase and squash them into the original one.

Also please follow the guidelines set in the PR template for the commit message:

```
Problem: x is broken

Solution: do y and z to fix it
```

**Pull request merged**